File: backend/apps/usuarios/repositories/horario_repository.py
from ..models import Horario


class HorarioRepository:

    @staticmethod
    def listar_todos():
        return Horario.objects.all()

    @staticmethod
    def buscar_por_id(horario_id):
        return Horario.objects.get(pk=horario_id)

    @staticmethod
    def activos_por_dia(profesional_id, dia):
        return Horario.objects.filter(
            profesional_id=profesional_id,
            dia=dia,
            activo=True
        )

    @staticmethod
    def existe_horario_valido(profesional_id, dia, hora):
        return Horario.objects.filter(
            profesional_id=profesional_id,
            dia=dia,
            activo=True,
            hora_inicio__lte=hora,
            hora_fin__gt=hora
        ).exists()

File: backend/src/repositories/horario.repository.ts
import { Horario } from "../models/Horario";

export class HorarioRepository {
  static listarTodos() {
    return Horario.find().lean();
  }
  static buscarPorId(id: string) {
    return Horario.findById(id);
  }
  static activosPorDia(profesionalId: string, dia: number) {
    return Horario.find({ profesional: profesionalId, dia, activo: true }).sort({ horaInicio: 1 });
  }
  static existeHorarioValido(profesionalId: string, dia: number, hora: string) {
    return Horario.exists({
      profesional: profesionalId,
      dia,
      activo: true,
      horaInicio: { $lte: hora },
      horaFin: { $gt: hora }
    });
  }
  static crear(data: any) {
    return Horario.create(data);
  }
}
